scipy/sparse/coo.py

```diff
@@ -190,6 +190,25 @@ def __init__(self, arg1, shape=None, dtype=None, copy=False):
 
         self._check()
 
+    def reshape(self, *args, order='C', copy=False):
+        shape = check_shape(args, self.shape)
+
+        nrows, ncols = self.shape
+
+        if order == 'C':
+            flat_indices = ncols * self.row + self.col
+            new_row, new_col = divmod(flat_indices, shape[1])
+        elif order == 'F':
+            flat_indices = self.row + nrows * self.col
+            new_col, new_row = divmod(flat_indices, shape[0])
+        else:
+            ValueError("'order' must be 'C' or 'F'")
+
+        return coo_matrix((self.data, (new_row, new_col)),
+                          shape=shape, copy=copy)
```

Comment:
The only thing that needs copying if `copy=True` is `self.data`, so I'd handle that here and pass `copy=False` to `coo_matrix`. Otherwise, you're making extra copies of `new_row` and `new_col`.

Reply:
This is a good idea, and something to keep in mind if more `reshape` methods get specialized.